consent-ws/src/main/java/org/broadinstitute/consent/http/resources/DataUseLetterResource.java

```diff
@@ -135,8 +135,8 @@ public Response getDUL(@PathParam("id") String consentId, @QueryParam("electionI
             if (StringUtils.isNotEmpty(consent.getLastElectionStatus())) {
                 election = api.retrieveElection(electionId, consentId);
             }
-            String fileUrl = election != null ? election.getDataUseLetter() : consent.getDataUseLetter();
-            String fileName = election != null ? election.getDulName() : consent.getDulName();
+            String fileUrl = (election != null && !election.getDataUseLetter().isEmpty()) ? election.getDataUseLetter() : consent.getDataUseLetter();
+            String fileName = (election != null && !election.getDulName().isEmpty()) ? election.getDulName() : consent.getDulName();
```

Comment:
when/how are we going to be backwards compatible? there probably needs to be a script to update existing elections. But once that has been done, there really should not be a case where an election does not have a DUL. Unless I am misunderstanding.

Reply:
You are right Andrea, these changes wouldn't be needed if we update existing elections. We will discard this PR and make a new one to include a MySql script.